app/src/main/java/camp/nextstep/edu/github/MainViewModelFactory.kt

```diff
@@ -0,0 +1,17 @@
+package camp.nextstep.edu.github
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.ViewModelProvider
+import camp.nextstep.edu.github.domain.GithubSearchRepository
+
+class MainViewModelFactory(
+    private val searchRepository: GithubSearchRepository,
+) : ViewModelProvider.Factory {
+    override fun <T : ViewModel> create(modelClass: Class<T>): T {
+        return if (modelClass.isAssignableFrom(MainViewModel::class.java)) {
+            MainViewModel(searchRepository) as T
+        } else {
+            throw IllegalArgumentException()
+        }
+    }
```

Comment:
depth를 최대한 줄이고 싶다면 같은 역할을 하는 구문을 다음처럼 작성할 수도 있습니다! 이 피드백은 참고만 해주세요 🙂 
```suggestion
    override fun <T : ViewModel> create(modelClass: Class<T>): T {
        require(modelClass.isAssignableFrom(MainViewModel::class.java)
        return MainViewModel(searchRepository) as T
    }
```

Reply:
require 쓰임새가 유용하네요~ 감사합니다! :)